### ISSUE TITLE
Updated Amazon VOD player

### ISSUE THREAD

**Issue by nroets1** [jurialmunkey/plugin.video.themoviedb.helper#657]
I found that the bundled amazon.json player fails to play tv shows with only one season. In order for Amazon VOD to work, I had to create 2 players: 
**my_amazon.json**
```
{
    "name"              : "Amazon Prime Video",
    "plugin"            : "plugin.video.amazon-test",
    "priority"          : 1,
    "is_resolvable"     : "true",
    "assert"            : {
                            "play_movie":       ["title", "year"],
                            "play_episode":     ["showname", "season", "episode"],
                            "search_movie":     ["title"],
                            "search_episode":   ["showname"]
                          },
    "play_movie"        : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{title}"},
                            {"label": "(?i).*{title}.*", "year": "{year}"}
                          ],
    "play_episode"      : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{showname}"},
                            {"title": "(?i).*{showname}.*"},
                            {"title": "(?i).*Season {season}"},
                            {"season": "{season}", "episode": "{episode}"}
                          ],
    "search_movie"      : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{title}"}
                          ],
    "search_episode"    : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{showname}"},
                            {"title": "(?i).*{showname}.*"},
                            {"title": "(?i).*Season {season}"}
                          ],
    "fallback"          : {"play_episode": "my_amazon-one.json play_episode"}
}

```
**my_amazon-one.json**
```
{
    "name"              : "Amazon Prime Video (One Season)",
    "plugin"            : "plugin.video.amazon-test",
    "priority"          : 100,
    "is_resolvable"     : "true",
    "assert"            : {
                            "play_movie":       ["title", "year"],
                            "play_episode":     ["showname", "season", "episode"],
                            "search_movie":     ["title"],
                            "search_episode":   ["showname"]
                          },
    "play_movie"        : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{title}"},
                            {"label": "(?i).*{title}.*", "year": "{year}"}
                          ],
    "play_episode"      : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{showname}"},
                            {"title": "(?i).*{showname}.*"},
                            {"season": "{season}", "episode": "{episode}"}
                          ],
    "search_movie"      : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{title}"}
                          ],
    "search_episode"    : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{showname}"},
                            {"title": "(?i).*{showname}.*"}
                          ]
}

```

**Comment by jurialmunkey** [comment]
Thanks. It's been a while since I used Amazon so I dont have a prime account currently.

I assume this is because shows with one season are flattened? Ie they open directly to episodes? If that's the case you should be able to handle them with a single player file using an early return.

Have a look at how the Netflix player does it. Specifically this line is the early return
https://github.com/jurialmunkey/plugin.video.themoviedb.helper/blob/ebb265e2a2bbdcae39907a9215655bd2dd8e6b4e/resources/players/netflix.json#L23

Basically it's a way to have two separate actions for the same step. In the Netflix example the return param tries to find a match for the episode instead of looking for a season folder. If it finds a matching episide it plays but if it doesn't it moves onto next line to open the season as if the return step didn't exist

**Comment by jurialmunkey** [comment]
Since I don't currently have an Amazon Prime account, can you test the following for your Play Episode command. I think it should allow you to have one single player for both scenarios.

```
"play_episode"      : [
    "plugin://plugin.video.amazon-test/",
    {"title": "(?i)search"},
    {"keyboard": "{showname}"},
    {"title": "(?i).*{showname}.*"},
    {"return": "true", "season": "{season}", "episode": "{episode}"},
    {"title": "(?i).*Season {season}"},
    {"season": "{season}", "episode": "{episode}"}
  ],
```

Once you confirm that it works, I'll update the built in player.

**Comment by nroets1** [comment]
Thank you. Working great. The {"return"} is not documented hence my approach but now I know! I've just changed it to use {"season:" "{season}"} after the {"return"} action instead of {"title": "(?i).*Season {season}"}. I've used {"title"} otherwise the first episode in a flattened show would always play. You may close this now.

**my_amazon.json**
```
{
    "name"              : "Amazon Prime Video",
    "plugin"            : "plugin.video.amazon-test",
    "priority"          : 1,
    "is_resolvable"     : "true",
    "assert"            : {
                            "play_movie":       ["title", "year"],
                            "play_episode":     ["showname", "season", "episode"],
                            "search_movie":     ["title"],
                            "search_episode":   ["showname"]
                          },
    "play_movie"        : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{title}"},
                            {"label": "(?i).*{title}.*", "year": "{year}"}
                          ],
    "play_episode"      : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{showname}"},
                            {"title": "(?i).*{showname}.*"},
                            {"return": "true", "season": "{season}", "episode": "{episode}"},
                            {"season": "{season}"},
                            {"season": "{season}", "episode": "{episode}"}
                          ],
    "search_movie"      : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{title}"}
                          ],
    "search_episode"    : [
                            "plugin://plugin.video.amazon-test/",
                            {"title": "(?i)search"},
                            {"keyboard": "{showname}"},
                            {"title": "(?i).*{showname}.*"},
                            {"title": "(?i).*Season {season}"}
                          ]
}

```
